allocator.py

```diff
@@ -5,17 +5,25 @@
 from kubernetes import client, config
 from kubernetes.client.rest import ApiException
 import os
+import glob
 
 # Read k8s config from environment KUBERNETES environment variables
 config.load_incluster_config()
 
 api_instance = client.CoreV1Api()
 name = os.environ["GPU_CONFIGMAP"]
 namespace = open("/var/run/secrets/kubernetes.io/serviceaccount/namespace").read()
+
+# Check for volume mounted GPUS first. Revert to environment variable.
+# Allow os.environ to throw if empty.
+visible_devices=','.join([os.path.basename(x) for x in glob.glob('/var/run/nvidia-container-devices/*GPU*')])
+if not visible_devices:
+    visible_devices=os.environ["NVIDIA_VISIBLE_DEVICES"]
```

Comment:
Wrapping with a slightly more explicit exception message _could_ save someone debugging a few extra brain cycles to parse the problem.

```suggestion
    try:
        visible_devices = os.environ["NVIDIA_VISIBLE_DEVICES"]
    except KeyError as e:
        raise ValueError("Expected NVIDIA_VISIBLE_DEVICES environment variable to be set") from e
```